mongomock/collection.py

```diff
@@ -826,6 +795,77 @@ def _copy_only_fields(self, doc, fields, container):
                 fields[field] = op
             return doc_copy
 
+    def _subdocitem_merge(self, doc_copy, field, sdocitems, subtract=False):
+        sdoc_copy = doc_copy.setdefault(field, [])
+        if len(sdoc_copy) < len(sdocitems):
+            sdoc_copy.extend([{} if not subtract else {'$EMPTY$': True}
+                              for _ in xrange(len(sdocitems) - len(sdoc_copy))])
+        for (sdoc, sdocitem) in zip(sdoc_copy, sdocitems):
+            if not subtract:
+                sdoc.update(sdocitem)
+            else:
+                if sdoc == {'$EMPTY$': True}:
+                    sdoc.clear()
+                    sdoc.update(sdocitem)
+                else:
+                    exc_keys = set(sdoc.keys()).difference(sdocitem.keys())
+                    for key in exc_keys:
+                        del sdoc[key]
+
+    def _filter_subdocuments(self, doc, fields, container):
+        # if 1 was passed in as the field values, include those fields
```

Comment:
I think you're expecting a bit much from a volunteer.

I'm more than happy to fix bugs or extend functionality that I've found useful for my own purposes, but to expect me to rewrite a portion of the existing code because the original code was poorly written seems a bit unfair.